src/app/client/projects/desktop/src/app/modules/user-profile/components/telemetry-import/telemetry-import.component.ts

```diff
@@ -0,0 +1,114 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { ResourceService, ToasterService } from '@sunbird/shared';
+import { ElectronDialogService } from '../../../offline/services';
+import { TelemetryActionsService } from './../../../offline/services';
+import * as _ from 'lodash-es';
+import { ActivatedRoute } from '@angular/router';
+import { TelemetryService } from '@sunbird/telemetry';
+import { timer, Subject, combineLatest } from 'rxjs';
+import { switchMap, map, filter, takeUntil } from 'rxjs/operators';
+@Component({
+  selector: 'app-telemetry-import',
+  templateUrl: './telemetry-import.component.html',
+  styleUrls: ['./telemetry-import.component.scss']
+})
+export class TelemetryImportComponent implements OnInit, OnDestroy {
+  importFilesList = [];
+  public unsubscribe$ = new Subject<void>();
+  importedFilesSize = 0;
+  callImportList = false;
+  apiCallSubject = new Subject();
+  contentStatusObject = {};
+  localStatusArr = ['inProgress', 'inQueue'];
+  apiCallTimer = timer(1000, 3000).pipe(filter(data => !data || (this.callImportList)));
+  constructor(public resourceService: ResourceService,
+    public telemetryActionsService: TelemetryActionsService,
+    private telemetryService: TelemetryService,
+    public toasterService: ToasterService,
+    private activatedRoute: ActivatedRoute,
+    private electronDialogService: ElectronDialogService) {
+      this.getList();
+      document.addEventListener('telemetry:import', (event) => {
+        this.apiCallSubject.next();
+      });
+     }
+
+  ngOnInit() {
+    this.apiCallSubject.next();
+  }
+  openImportTelemetryDialog() {
+    this.setImportTelemetry();
+    this.electronDialogService.showTelemetryImportDialog();
+  }
+  getList() {
+    // tslint:disable-next-line: deprecation
+    combineLatest(this.apiCallTimer, this.apiCallSubject, (data1, data2) => true)
+      .pipe(takeUntil(this.unsubscribe$),  switchMap(() => this.telemetryActionsService.telemetryImportList()),
+        map((resp: any) => {
+          this.callImportList = false;
```

Comment:
use tap to check the data and use _.find method  to reduce the number of lines

Reply:
const itemToComplete =  _.find(_.get(resp, 'result.response.items'), (item) => return ['inProgress', 'inQueue'].includes(item.status))
this. callImportList  =  itemToComplete ? true: false